docs/rfcs/0023-samhsa-v2.md

```diff
@@ -0,0 +1,218 @@
+# RFC Proposal
+
+* RFC Proposal ID: `0023-samhsa-v2`
+* Start Date: 2024-10-11
+* RFC PR: [beneficiary-fhir-data/rfcs#0023](https://github.com/CMSgov/beneficiary-fhir-data/pull/2464)
+* JIRA Ticket(s):
+    * [BFD-3660](https://jira.cms.gov/browse/BFD-3660)
+
+A proposal to rewrite the SAMHSA filter to be more reliable and easier to maintain.
+
+## Status
+
+* Status: Proposed <!-- (Proposed/Approved/Rejected/Implemented) -->
+* Implementation JIRA Ticket(s):
+
+## Table of Contents
+
+- [RFC Proposal](#rfc-proposal)
+  - [Status](#status)
+  - [Table of Contents](#table-of-contents)
+  - [Background](#background)
+  - [Implementation Overview](#implementation-overview)
+  - [Database Schema](#database-schema)
+    - [Optional Addition: Claim Details](#optional-addition-claim-details)
+    - [Entity Definitions - DSL vs JPA](#entity-definitions---dsl-vs-jpa)
+  - [Pipeline Updates](#pipeline-updates)
+  - [FHIR Server Updates](#fhir-server-updates)
+
+## Background
+
+The SAMHSA filter currently operates on the transformed representation of the FHIR payloads.
+It looks at the diagnosis, procedure, and detail components of the payload and checks the list of ICD, CPT, and DRG codes. 
+When `excludeSAMHSA=true` is supplied to the EOB, Claim, or ClaimResponse endpoints, any claim that contains at least one code that matches against the SAMHSA list will be excluded.
+
+The current filter logic is tightly coupled to the implementation of the FHIR transformers.
+Any change to the system URLs, additional functionality that adds a potentially sensitive code to a new field in the FHIR response, or a bug that causes certain fields to not be mapped in the FHIR payload may cause the filter to behave incorrectly. 
+
+Instead of basing the SAMHSA filter on the FHIR payload, it would be safer to check any database column that could contain sensitive info and explicitly mark that claim as sensitive at ingestion time.
+
+## Implementation Overview
+
+The implementation will utilize [FHIR Security Labels](https://build.fhir.org/security-labels.html) to tag claims as 'sensitive'.
+SAMHSA claims will include a [Confidentiality](https://terminology.hl7.org/4.0.0/CodeSystem-v3-Confidentiality.html) tag of `R` for 'restricted' and an [Act Code](https://terminology.hl7.org/6.0.2/CodeSystem-v3-ActCode.html) tag of `42CFRPart2`.
```

Comment:
I think the content is nearly or entirely identical, but it's important that we clearly refer to the like-sequenced FHIR™ R4 and/or 4.0.1 where possible.
```suggestion
The implementation will utilize [FHIR Security Labels](https://hl7.org/fhir/R4/security-labels.html) to tag claims as 'sensitive'.
SAMHSA claims will include a [Confidentiality](https://terminology.hl7.org/5.5.0/CodeSystem-v3-Confidentiality.html#v3-Confidentiality-R) tag of `R` for 'restricted' and an [Act Code](https://terminology.hl7.org/5.5.0/CodeSystem-v3-ActCode.html#v3-ActCode-42CFRPart2CD) tag of `42CFRPart2`.
```

Reply:
Good catch, thanks!